streamer/src/nonblocking/quic.rs

```diff
@@ -146,21 +147,45 @@ async fn setup_connection(
                 let stake = *stake;
                 let total_stake = staked_nodes.total_stake;
                 drop(staked_nodes);
-                let mut connection_table_l = staked_connection_table.lock().unwrap();
-                if connection_table_l.total_size >= max_staked_connections {
-                    let max_connections = max_percentage_full.apply_to(max_staked_connections);
-                    let num_pruned = connection_table_l.prune_oldest(max_connections);
-                    stats.num_evictions.fetch_add(num_pruned, Ordering::Relaxed);
-                }
+
                 connection.set_max_concurrent_uni_streams(
```

Comment:
I can see that perspective as well but I think it's clearer to have every connection in the unstaked connection table to all be limited to `QUIC_MAX_UNSTAKED_CONCURRENT_STREAMS`.

Reply:
I think, it'll be opposite of the QoS we are trying to provide for the staked peers. A peer with a lower stake might be able to setup more streams than a peer with a higher stake (because we could not fit the latter in the staked connection table).

Maybe, the logic could be changed to look at the peer stake instead of which table it's being put into. And, that check can be moved inside `try_add_connection`.

Reply:
> A peer with a lower stake might be able to setup more streams than a peer with a higher stake (because we could not fit the latter in the staked connection table).

I noticed this while reviewing as well and just commented here to discuss this strange behavior:

https://github.com/solana-labs/solana/pull/25993#issuecomment-1171460550

Reply:
Yes, the comment makes sense to me. 

Considering a few use cases, RPC nodes will open a lot of streams on a given connection. And may not have any stake.
On the flip side, transaction forwarding will also open a lot of streams, and will potentially have a lot of stake (as the node probably was the previous leader).

Reply:
I think we need to overhaul the calculation of maximum uni_streams allowed for a connection.
In the current PR I have cleaned up the code so that it's at one place. I'll file an issue to track the problem, but I think it's better if we have a standalone PR to fix it.

Reply:
Created this issue to track it: https://github.com/solana-labs/solana/issues/26340

Reply:
I agree that staked / unstaked stream limits can be fixed elsewhere but I still think that any connection in the unstaked connection table should be limited to the max unstaked stream limit const